pyproject.toml

```diff
@@ -42,7 +42,7 @@ rich = ">=12.2.0, <14.0.0"
 tzlocal = ">=4.0"   # https://github.com/regebro/tzlocal/blob/master/CHANGES.txt
 
 [tool.poetry.dev-dependencies]
-black = { version = ">=21.5b2", allow-prereleases = true }
+black = { version = ">=23.11.0", allow-prereleases = true }
```

Comment:
Two tests were failing (Mac-latest Python 3.11, 3.12) because they used a version of black that didn't support the Apple Silicon architecture (M1, M2 chips). Starting from version 23.11.0, black resolves this compatibility issue. For more details, refer to [this issue](https://github.com/psf/black/issues/3981).

Reply:
Thanks for figuring this out, @eigenric. Looks like the poetry lockfile just needs to be updated to lock in that latest version. I've merged that change into `develop` (#1890). Could you merge `develop` into this PR? That should fix the issue and finally get these tests running.